### PR TITLE
Js2

### PR DESCRIPTION
**Pull request by chasju**
Teacher Review

### CODE REVIEW

**Comment by Fermain** [comment]
UX - Redirect to post page when updating post

**Comment by Fermain** [comment]
Improvement - Use profile data and placeholder images instead of hardcoding this data

**Comment by Fermain** [comment]
Improvement - Use `aspect-ratio: 1` to keep images perfectly round.

File: src/js/handlers/createPost.mjs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    const bodyText = {
+      title: "No title",
+      body: input.value,
+    };
+    createPost(bodyText);
+    input.value = "";
+
+    if (location.pathname === "/pages/profile/") {
+      setTimeout(getProfileFeedPosts, 300);
```
Comment (Fermain) [inline src/js/handlers/createPost.mjs:27]: This is a sign that something is wrong. Why is this in use?

File: src/js/handlers/deletePost.mjs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+export function deletePost() {
+  const deleteButton = document.querySelectorAll(".delete-post");
+
+  deleteButton.forEach((button) => {
+    button.addEventListener("click", (e) => {
+      removePost(e.target.id);
+      if (location.pathname === "/pages/profile/") {
+        setTimeout(getProfileFeedPosts, 300);
```
Comment (Fermain) [inline src/js/handlers/deletePost.mjs:18]: Same issue here, what is the purpose of the timeout?

File: src/js/handlers/searchPosts.mjs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      e.preventDefault();
+      const filterPosts = posts.filter((post, i, posts) => {
+        const handle = post.author.name.toLowerCase();
+        const body = post.body.toLowerCase();
+
+        if (handle.includes(searchInput.value.toLowerCase()) || body.includes(searchInput.value.toLowerCase())) {
+          return post;
+        }
+        if (searchInput === "") {
+          return posts;
```
Comment (Fermain) [inline src/js/handlers/searchPosts.mjs:38]:
```suggestion
          return true;
```

File: src/js/handlers/searchPosts.mjs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    const container = document.querySelector(".post-container");
+
+    searchForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      const filterPosts = posts.filter((post, i, posts) => {
+        const handle = post.author.name.toLowerCase();
+        const body = post.body.toLowerCase();
+
+        if (handle.includes(searchInput.value.toLowerCase()) || body.includes(searchInput.value.toLowerCase())) {
+          return post;
```
Comment (Fermain) [inline src/js/handlers/searchPosts.mjs:35]:
```suggestion
          return true;
```

Comment (Fermain) [reply src/js/handlers/searchPosts.mjs:35]: Filter function only accepts a boolean return type

File: src/js/handlers/searchPosts.mjs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+  try {
+    const posts = await getPosts();
+    const user = localStorage.getItem("isLoggedIn");
+    const searchInput = document.querySelector("#searchProfilePage");
+    const searchForm = document.querySelector("#searchForm");
+
+    const filteredPosts = posts.filter((post) => {
+      if (JSON.stringify(post.author.name) === user) {
+        return post;
```
Comment (Fermain) [inline src/js/handlers/searchPosts.mjs:81]:
```suggestion
        return true;
```

File: src/js/handlers/searchPosts.mjs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    });
+
+    searchForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      const filterPosts = filteredPosts.filter((post, i, filteredPosts) => {
+        const handle = post.author.name.toLowerCase();
+        const body = post.body.toLowerCase();
+
+        if (handle.includes(searchInput.value.toLowerCase()) || body.includes(searchInput.value.toLowerCase())) {
+          return post;
```
Comment (Fermain) [inline src/js/handlers/searchPosts.mjs:92]:
```suggestion
          return true;
```

File: src/js/handlers/searchPosts.mjs [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        if (searchInput === "") {
+          return filteredPosts;
```
Comment (Fermain) [inline src/js/handlers/searchPosts.mjs:95]:
```suggestion
          return true;
```

**Review by Fermain — APPROVED** [review]
This is a good submission with some areas of improvement around User Experience, and some corrections to be made around `array.filter`. These do not change the functionality of the site, therefore have nothing to do with your grade - however I do recommend that you review the content on filter to make sure you are using it correctly in future.

Overall your code is well organised and split up into manageable chunks using the module system. The UI is clear and simple, providing plenty of options and components to the user. It would be great to see more functionality added using your stable approach.

File: src/js/handlers/createPost.mjs (hunk 1 above)
Comment (chasju) [reply src/js/handlers/createPost.mjs:27]: Resolved. 
Made the event listener an async function and awaited the get posts functions.

File: src/js/handlers/deletePost.mjs (hunk 2 above)
Comment (chasju) [reply src/js/handlers/deletePost.mjs:18]: Resolved. 
Made the event listener an async function and awaited the get posts functions.

File: src/js/handlers/searchPosts.mjs (hunk 3 above)
Comment (chasju) [reply src/js/handlers/searchPosts.mjs:38]: Resolved.
Removed the second if condition, and returned true instead of 'post'.

**Pull request merged**